File: materials/2019-2020/11v/2020-02-20-json-encoder/Point.java
class Point {
  private double x;
  private double y;

  public Point() {
    x = 0;
    y = 0;
  }

  public Point(double x, double y) {
    this.x = x;
    this.y = y;
  }

  public Point(double x) {
    this.x = x;
    y = 0;
  }

  public double getX() {
    return x;
  }

  public double getY() {
    return y;
  }

  public String toString() {
    return String.format("(%f, %f)", getX(), getY());
  }
}

File: materials/2021-2022/11a/2022-01-05-abstract/point.cpp
#include "point.hh"
#include <iostream>
#include <sstream>

using namespace std;
Point::Point(int x, int y) { this->x = this->y; }
Point::Point() {
  x = 0;
  y = 0;
}

void Point::set_x(int x) { this->x = x; }
void Point::set_y(int y) { this->y = y; }

int Point::get_x() const { return x; }
int Point::get_y() const { return y; }

void Point::print() const {
    cout << "Point(" << x << ", " << y << ")";
}

string Point::to_string() const {
    ostringstream o;
    o << "Point(" << x << ", " << y << ")";
    return o.str();
}